src/fiat/funds-transfers.md

```diff
@@ -57,13 +57,14 @@ curl -X POST "https://service.centrapay.com/api/topups" \
 
 **Error Responses**
 
-| Status |             Code              |                         Description                         |
-| :----- | :---------------------------- | :---------------------------------------------------------- |
-| 403    | {% break _ BANK_AUTHORITY_WALLET_MISMATCH %} | The wallet and the bank authority for the top up request do not belong to the same account. |
-| 403    | {% break _ MAX_INFLIGHT_TOPUPS_EXCEEDED %}   | The bank authority already has ten pending top ups, which is the maximum a bank authority can have at any one time. |
+| Status |                       Code                       |                                                                                                             Description                                                                                                              |
+| :----- | :----------------------------------------------- | :----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| 403    | {% break _ BANK_AUTHORITY_WALLET_MISMATCH %}     | The wallet and the bank authority for the top up request do not belong to the same account.                                                                                                                                          |
+| 403    | {% break _ MAX_INFLIGHT_TOPUPS_EXCEEDED %}       | The bank authority already has ten pending top ups, which is the maximum a bank authority can have at any one time.                                                                                                                  |
 | 403    | {% break _ MAX_INFLIGHT_TOPUP_AMOUNT_EXCEEDED %} | The top up can not be created because it is above the max amount of funds a bank authority can have pending at any one time. The max amount is $1000.00 for verified bank authorities and $100.00 for non-verified bank authorities. |
+| 403    | {% break _ MISSING_BANK_AUTHORITY_INFORMATION %} | Bank authority is missing required fields for topup. See [bank authorities](bank-authorities.html#bank-fields-anchor) documentation for a list of required fields.                                                                   |
```

Comment:
I think this should be DIRECT_DEBIT_NOT_AUTHORIZED. A bank account should be authorized or not for DD and not some state in between. Every time DD auth is added to a bank account then the identifying params should be provided.

Reply:
Are you suggesting that our update endpoint should force all 3 fields to be submitted together?

Reply:
Our intention is to roll out this:

![Add a new bank account from withdrawl](https://user-images.githubusercontent.com/45738/112917582-72ddef00-915f-11eb-9365-86d80e4270e4.png)

The problem with this screen is that it lacks `userAuthorizesDirectDebit`, `fullName` and `emailAddress` and all 3 optional fields would be required for topup.

Calling this `DIRECT_DEBIT_NOT_AUTHORIZED` implies you might be able to do this without fullName` and `emailAddress` filled out.

I'd prefer to keep with `MISSING_BANK_AUTHORITY_INFORMATION` to nod to all optional fields and make our integrations check against `topupFieldsRequired`.

Reply:
Other required values are picked from Cognito User Profile. UI could check if those values are present and ask for them if they are not.

Reply:
I've talked this over with @maticko, and here are the options we've explored:

1. Breaking change
    - force `DIRECT_DEBIT_NOT_AUTHORIZED`
        - have 3 fields you have to specify together
        - this one is good, but we're not keen on making a non-breaking change
    - keep `MISSING_BANK_AUTHORITY_INFORMATION`
        - allow any field to be updated anytime (but do it together in our UX)
        - have 3 fields you can update anytime till they're verified
        - this won't work with our S3 plan, so it's a no go

2. Non breaking change
    - create a new endpoint and deprectate the old
      i.e. https://service.centrapay.com/api/accounts/${accountId}/bank-authorities
    - force `DIRECT_DEBIT_NOT_AUTHORIZED`
        - have 3 fields you have to specify together

We're feeling pretty good about number 2 and I'll update the docs to reflect this. If you can think of another angle or a different way, please sing out.

Reply:
This is now updated such that

- We have added a DEPRECATED flag to existing create endpoint
- We have a new API endpoint for creating bank authorities
- We have a new API endpoint for finalizing a bank authority
- PII is removed from all bank authority responses
- DIRECT_DEBIT_NOT_AUTHORIZED is thrown from topup create endpoint
- directDebitAuthorized has been added to all API responses
- topupFieldsRequired has been removed from all API responses

I believe this covers our requirements. Please have another review and tell me what-cha think :)